### DISCUSSION TITLE
Nuxt 3 override runtimeConfig at runtime based on the domain (multidomain sites)

### DISCUSSION THREAD

**Discussion by salomonsanz** [nuxt/nuxt#16516] [Questions]
I have a multidomain site in several countries with nuxt 3.

I'm trying to modify the _runtimeConfig_ at runtime execution based on the current request domain, is that possible?

For example, if the user is in mydomain.es

```
{
public : {
   googleanalytics_id : 'xxxxx'
}
```

For example, if the user is in mydomain.fr
```
{
public : {
   googleanalytics_id : 'yyyyyyy'
}
```

**Comment by salomonsanz** [comment]
I finally found a way to do this, it is not very elegant, but it works...

at: _plugins/configByDomain.server.ts_
```
import { loadConfig } from '../config/SiteConfig'

export default defineNuxtPlugin(async nuxtApp => {
    const domain = useRequestHeaders().host.split(':')[0];
    const SiteConfig = await loadConfig(domain);        
    // to make the object writable
    nuxtApp.ssrContext.runtimeConfig = JSON.parse(JSON.stringify(nuxtApp.ssrContext.runtimeConfig));    
    Object.assign(nuxtApp.ssrContext.runtimeConfig, SiteConfig);
})
```

**Reply by MarioC3** [reply]
Hi @salomonsanz!

Is this the only solution you've encountered? 
Is there no mechanism that Nuxt allows to override runtimeConfigs on plugins? 

I've just been looking all over the repo, but I couldn't find a solution other than this.

**Reply by salomonsanz** [reply]
Hi Mario, my solution only works for pure SSR mode, I couldn't find a way to override the payload...
Sorry, I'm looking for a solution for this too.

**Reply by andysay** [reply]
any update for this ? :)

**Comment by zailaib** [comment]
i want switch dbName with mongoose for tenant isolation 

![image](https://github.com/nuxt/nuxt/assets/10792685/0c81687f-a65a-42ab-884e-e33f5cac6ff1)

not work for me.
